src/core/parser.ts

```diff
@@ -35,13 +36,73 @@ const formatExtensions: Record<DiagnosticsFormat, string> = {
 
 const validFormats = ['stylish', 'json', 'junit', 'html', 'text', 'teamcity', 'pretty'];
 
+type AuthEntry = {
+  pattern: string;
+  token: string;
+};
+
+const CONFIG_FILE = path.join(os.homedir(), '.asyncapi', 'config.json');
+
+function loadAuthEntries(): AuthEntry[] {
+  if (!fs.existsSync(CONFIG_FILE)) { return [];}
+  const content = fs.readFileSync(CONFIG_FILE, 'utf8');
+  try {
+    const parsed = JSON.parse(content);
+    return Array.isArray(parsed?.auth) ? parsed.auth : [];
+  } catch (e) {
+    console.error('❌ Invalid JSON in config file:', e);
+    return [];
+  }
+}
+
+function normalizeRawGitUrl(uri: URI): string {
+  // Converts: raw.githubusercontent.com/org/repo/branch/path → https://github.com/org/repo/tree/branch
+  const segments = uri.path().split('/').filter(Boolean); // Remove empty segments
+  if (segments.length < 3) { return uri.toString(); }
+
+  const [org, repo, branch] = segments;
+  return `https://github.com/${org}/${repo}/tree/${branch}`;
+}
+
+const customResolver = {
+  schema: 'https',
+  order: 1,
+
+  canRead: (uri: URI) => uri.hostname() === 'raw.githubusercontent.com',
```

Comment:
Why checking for GitHub here? 🤔 References can be in many places, not just GitHub, and even in such a case the GitHub deployment might be in a private domain, which is pretty common in medium to large companies.